code/datums/traits/negative.dm

```diff
@@ -619,13 +621,14 @@
 	if(world.time > next_process)
 		next_process = world.time + process_interval
 		var/deleted = QDELETED(reagent_instance)
-		if(deleted || !LAZYFIND(H.reagents.addiction_list, reagent_instance))
-			if(deleted)
-				reagent_instance = new reagent_type()
-			else
-				reagent_instance.addiction_stage = 0
-			H.reagents.addiction_list += reagent_instance
-			to_chat(quirk_holder, "<span class='danger'>You thought you kicked it, but you suddenly feel like you need [reagent_instance.name] again...</span>")
+		if(H.reagents.addiction_list)
```

Comment:
You shouldn't need to have this, `LAZYFIND` checks if the list is null.

Reply:
From my testing, it runtimed. Might have been `H.reagents.addiction_list += reagent_instance` causing it, though.

Reply:
That's more likely to me, try replacing that with another `LAZYADD` and give it a shot.